packages/backend/src/redirect.ts

```diff
@@ -0,0 +1,67 @@
+import { parsePublishableKey } from './util/parsePublishableKey';
+
+type RedirectAdapter = (url: string) => any;
+
+type SingUpParams = { url?: string; returnBackUrl?: string };
+type SingInParams = { url?: string; returnBackUrl?: string };
+
+const buildPath = (destinationUrl: string, currentUrl?: string) => {
+  let destinationURL;
+  if (!destinationUrl.startsWith('http')) {
+    if (!currentUrl || !currentUrl.startsWith('http')) {
+      throw new Error('destination url or return back url should be an absolute path url!');
+    }
+
+    const currentURL = new URL(currentUrl);
+    destinationURL = new URL(destinationUrl, currentURL.origin);
+  } else {
+    destinationURL = new URL(destinationUrl);
+  }
+
+  if (currentUrl) {
+    destinationURL.searchParams.set('redirect_url', currentUrl);
+  }
+
+  return destinationURL.toString();
+};
+
+type RedirectParams = {
+  redirectAdapter: RedirectAdapter;
+  signInUrl?: string;
+  signUpUrl?: string;
+  publishableKey?: string;
+  frontendApi?: string;
+};
+
+export function redirect({ redirectAdapter, signUpUrl, signInUrl, frontendApi, publishableKey }: RedirectParams) {
+  const accountsBaseUrl = getAccountsBaseUrl(frontendApi, publishableKey);
```

Comment:
🙃 It's more reasonable to do the following:

1. Check if frontendApi is passed
2. If yes, pass it to buildAccountsBaseUrl(frontendApi)
3. If not passed, check if publishableKey is passed
4. If yes, parse it and pass it to buildAccountsBaseUrl(frontendApi)
5. Otherwise the accountsBaseUrl should be "" so that you can easily throw an error later if signIn or signUp URLs are missing.

Reply:
I will refactor it. Are you also suggesting to move the check that `signUpUrl` / `sigInUrl` or `frontentApi`/`publishableKey` before the execution of the `redirectToSignIn` / `redirectToSignUp` and enforce that either both `signUpUrl` and `signInUrl` or `frontentApi`/`publishableKey` should be defined ?

Reply:
The keys should exist at this point but we can rethrow the missing key error from the errorthrower to be on the safe side

Reply:
i'm gonna keep the error being thrown upon helper execution and change the error message to be the same as `errorThrower.throwMissingPublishableKeyError()` from `@clerk/shared`. We currently do not have `@clerk/shared` as dependency due to some bundling issues so i will leave a TODO to replace that errorMessage with the errorThrower when `@clerk/shared` is added.